From 8dd6521980f8b6c8f638b8cf1948c2f863ce040a Mon Sep 17 00:00:00 2001
From: Sammy He <r62914@freescale.com>
Date: Wed, 9 Nov 2011 23:03:04 +0800
Subject: [PATCH 171/259] ENGR00161765 vpu: Fix the wrong code in setting
 BIT_FRAME_MEM_CTRL reg

Fix the wrong code in setting BIT_FRAME_MEM_CTRL register. This code
may cause decoded output color space not correct.

Signed-off-by: Sammy He <r62914@freescale.com>
---
 vpu/vpu_lib.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/vpu/vpu_lib.c b/vpu/vpu_lib.c
index 9beb8b9..f0e0436 100644
--- a/vpu/vpu_lib.c
+++ b/vpu/vpu_lib.c
@@ -2462,7 +2462,7 @@ RetCode vpu_DecOpen(DecHandle * pHandle, DecOpenParam * pop)
 	UnlockVpuReg(vpu_semap);
 
 	val = VpuReadReg(BIT_FRAME_MEM_CTRL);
-	val &= ~(1 << 2 || 1 << 3); /* clear the bit firstly */
+	val &= ~(1 << 2 | 1 << 3); /* clear the bit firstly */
 	val &= 0x3f;
 	if (cpu_is_mx6q()) {
 		if (pDecInfo->openParam.bitstreamMode)
-- 
1.8.0

